internal/signerverifier/ssh/ssh.go

```diff
@@ -0,0 +1,164 @@
+	Public string
+}
+
+func (k *Key) Verify(ctx context.Context, data []byte, sig []byte) error {
+
+	pub, err := parseSSH2Body(k.KeyVal.Public)
+	if err != nil {
+		return fmt.Errorf("failed to parse ssh public key material: %v", err)
+	}
+
+	signature, err := sshsig.Unarmor(sig)
+	if err != nil {
+		return fmt.Errorf("failed to parse ssh signature: %v", err)
+	}
+
+	message := bytes.NewReader(data)
+
+	// ssh-keygen uses sha512 to sign with **any*** key
+	hash := sshsig.HashSHA512
+	if err = sshsig.Verify(message, signature, pub, hash, "gittuf"); err != nil {
+		return fmt.Errorf("failed to verify ssh signature: %v", err)
+	}
+
+	return nil
+}
+
+func (k *Key) KeyID() (string, error) {
+	return k.keyID, nil
+}
+
+// FIXME: required by dsse Verifier interface; consider fixing interface
+func (k *Key) Public() crypto.PublicKey {
+	sshKey, _ := parseSSH2Body(k.KeyVal.Public)
+	return sshKey.(ssh.CryptoPublicKey).CryptoPublicKey()
+}
+
+// dsse.Signer interface implementation for ssh key signing
+// Includes `Key` as single source of truth for signing scheme and keyid
+type Signer struct {
+	Key  *Key
+	Path string
+}
+
+func (s *Signer) Sign(ctx context.Context, data []byte) ([]byte, error) {
+	cmd := exec.Command("ssh-keygen", "-Y", "sign", "-n", "gittuf", "-f", s.Path)
+
+	cmd.Stdin = bytes.NewBuffer(data)
+
+	output, err := cmd.Output()
+	if err != nil {
+		// TODO: Handle signing error (exit 255) on world-readable key file
+		//    git doesn't store read permissions; maybe need to chmod in test
+		return nil, fmt.Errorf("failed to run command %v: %v", cmd, err)
+	}
+
+	return output, nil
+
+}
+
+func (s *Signer) KeyID() (string, error) {
+	return s.Key.keyID, nil
+}
+
+// Import Key from path using ssh-keygen
+// Path can be public or private encrypted or plain key, much like git's
+// user.signingKey config
+func Import(path string) (*Key, error) {
+	cmd := exec.Command("ssh-keygen", "-e", "-f", path)
+	output, err := cmd.Output()
+	if err != nil {
+		return nil, fmt.Errorf("failed to run command %v: %v", cmd, err)
+	}
+
+	sshPub, err := parseSSH2Key(string(output))
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse SSH2 key: %v", err)
+	}
+
+	return &Key{
+		keyID:   ssh.FingerprintSHA256(sshPub),
+		KeyType: "ssh",
+		Scheme:  sshPub.Type(),
+		KeyVal: KeyVal{
+			Public: base64.StdEncoding.EncodeToString(sshPub.Marshal()),
+		},
+	}, nil
+}
+
+func parseSSH2Body(body string) (ssh.PublicKey, error) {
+	bodyBytes, err := base64.StdEncoding.DecodeString(body)
+	if err != nil {
+		return nil, err
+	}
+	return ssh.ParsePublicKey(bodyBytes)
+}
+
+// Parse SSH2 public key as defined in RFC4716 (3.  Key File Format)
+// Ignores:
+// - line length constraints
+// - other line termination charcters than "\n"
+// - exact header tag and header value format
+func parseSSH2Key(data string) (ssh.PublicKey, error) {
+
+	header := "---- BEGIN SSH2 PUBLIC KEY ----"
+	footer := "---- END SSH2 PUBLIC KEY ----"
+	lineSep := "\n"
+	commentSep := ":"
```

Comment:
nit: I'd make this more generic as there a handful of defined headers + support for extended headers that a user may put in there separately?

Reply:
Actually, can ssh-keygen accept a pubkey already formatted as rfc4716 and still "convert" to rfc4716? I ran into a little trouble first with the permissions being too open for a private key (despite passing in a public key) and then a libcrypto error (after setting perms to 600). Can we handle a case where a user has an RFC4716 encoded key on disk?

Reply:
> nit: I'd make this more generic as there a handful of defined headers + support for extended headers that a user may put in there separately?

What do you mean by "make this more generic"? Would you like to parse out the headers? I don't think we need them, do we?

Reply:
purely in the var name: it's a separator for all headers, not just `comment`. as I said, it's a nit that shouldn't block this :D

Reply:
> Can we handle a case where a user has an RFC4716 encoded key on disk?

Do you mean to load the key from file? Sure, we can do all sorts of things. But I find it quite attractive to just ask `ssh-keygen` for the key, because it hides so much complexity.

Reply:
![image](https://github.com/gittuf/gittuf/assets/8928778/abfb0c86-cc5e-4935-afeb-a9602030885c)

Possibly I'm doing something wrong? :sweat_smile:

Reply:
I think it means that `ssh-keygen` cannot read rfc4716 keys.

Reply:
Why do you want to support rfc4716 keys?